frontend/packages/console-dynamic-plugin-sdk/src/extensions/create-resource.ts

```diff
@@ -0,0 +1,30 @@
+import { match as RouterMatch } from 'react-router';
+import { ExtensionK8sModel } from '../api/common-types';
+import { CodeRef, Extension, ExtensionDeclaration } from '../types';
+
+export type CreateResource = ExtensionDeclaration<
+  'console.resource/create',
+  {
+    /** The model for which this create resource page will be rendered. */
+    model: ExtensionK8sModel;
+    /** The component to be rendered when the model matches */
+    component: CodeRef<
+      React.ComponentType<{
+        match: RouterMatch<{
+          /** Current namespace */
+          ns?: string;
+          /**
+           * Name of the OLM Operator ClusterServiceVersion that manages this resource.
+           * Defined if resource creation is initiated from the Operator details page.
+           */
+          csvName?: string;
```

Comment:
I don't think this param should belong in such a generic extension.

Reply:
`csvName` is currently used in the OLM form to get the actual `csv` resource and load some additional data from it. Im not sure if the parameter is really needed for plugins.

@TheRealJon @spadgett WDYT ?

Reply:
Having the contributor need to render a different header to remain consistent will be problematic. Unless there's a component that can be reused and it can use the route info from a hook instead of relying on it being passed in as a prop.

This is why I suggested we render the headers and then use the extension to render the form area only.

Reply:
@rawagner can you render the operands breadcrumbs in OLM before rendering the extension component? This would make it so that contributors don't need to worry about the rendering of breadcrumbs.

Reply:
`csvName` might not be specific enough if this is a cluster-scoped resource since `ns` would be undefined.

I think in practice this isn't needed. Presumably the same operator that contributed the custom resource is customizing the creation page using a plugin.

Reply:
@rawagner @spadgett @christianvogt I think we probably won't even need the CSV for a custom Operand creation form. The only thing we use it for in the current creation form is to get OLM descriptors and display operator details in the sidebar. Descriptors become obsolete in the context of a custom creation form, and an operator author can provide whatever Operator info they want on their custom creation form.

It also makes sense to me that the user should only need to implement the main content of the page, excluding the header and breadcrumbs. Maybe we could implement extensions just for overriding page headers or breadcrumbs if it becomes an issue?

Reply:
It seems we want to surface the same form in different contexts (operand and details page, maybe even the catalog in the future). Definitely the extensions shouldn't need to worry about the breadcrumbs.